lumicks/pylake/detail/confocal.py

```diff
@@ -15,12 +16,18 @@
 from .mixin import ExcitationLaserPower, PhotonCounts
 from .utilities import could_sum_overflow
 
-linear_colormaps = {
-    "red": LinearSegmentedColormap.from_list("red", colors=[(0, 0, 0), (1, 0, 0)]),
-    "green": LinearSegmentedColormap.from_list("green", colors=[(0, 0, 0), (0, 1, 0)]),
-    "blue": LinearSegmentedColormap.from_list("blue", colors=[(0, 0, 0), (0, 0, 1)]),
-    "rgb": None,
-}
+
+ColorMaps = namedtuple("ColorMaps", ["red", "green", "blue", "magenta", "yellow", "cyan", "rgb"])
+
+colormaps = ColorMaps(
+    red=LinearSegmentedColormap.from_list("red", colors=[(0, 0, 0), (1, 0, 0)]),
+    green=LinearSegmentedColormap.from_list("green", colors=[(0, 0, 0), (0, 1, 0)]),
+    blue=LinearSegmentedColormap.from_list("blue", colors=[(0, 0, 0), (0, 0, 1)]),
+    magenta=LinearSegmentedColormap.from_list("magenta", colors=[(0, 0, 0), (1, 0, 1)]),
+    yellow=LinearSegmentedColormap.from_list("yellow", colors=[(0, 0, 0), (1, 1, 0)]),
+    cyan=LinearSegmentedColormap.from_list("cyan", colors=[(0, 0, 0), (0, 1, 1)]),
+    rgb=None,
+)
```

Comment:
A suggestion to solve the issue with the documentation of the API: You could create a class inheriting from the namedtuple, add a docstring to the class and overwrite the docstrings of the fields:
```python3
_maps = {
    "red": LinearSegmentedColormap.from_list("red", colors=[(0, 0, 0), (1, 0, 0)]),
    "green": LinearSegmentedColormap.from_list("green", colors=[(0, 0, 0), (0, 1, 0)]),
    "blue": LinearSegmentedColormap.from_list("blue", colors=[(0, 0, 0), (0, 0, 1)]),
    "magenta": LinearSegmentedColormap.from_list("magenta", colors=[(0, 0, 0), (1, 0, 1)]),
    "yellow": LinearSegmentedColormap.from_list("yellow", colors=[(0, 0, 0), (1, 1, 0)]),
    "cyan": LinearSegmentedColormap.from_list("cyan", colors=[(0, 0, 0), (0, 1, 1)]),
}


class ColorMaps(namedtuple("ColorMaps", _maps.keys())):
    """ColorMaps to plot with Pylake"""

for key in _maps.keys():
    attr = getattr(ColorMaps, key)
    attr.__doc__ = f"{key} color map for plotting"

colormaps=ColorMaps(**_maps)
```
<img src="https://user-images.githubusercontent.com/28353120/204365468-8bcf65a4-d64f-44d0-bcad-8e176f8b9ea1.png" width="250">


Drawback is that the fields are not sorted and also other methods of `namedtuple` are documented. Alternatively, I like the cleaner code and concise output of dataclasses:
```python3
class ColorMaps(
    make_dataclass(
        "ColorMaps", [(key, LinearSegmentedColormap) for key in _maps.keys()], frozen=True
    )
):
    """ColorMaps to plot with Pylake"""
```
<img src="https://user-images.githubusercontent.com/28353120/204365769-ded6a729-ae4b-45f4-8fcf-7ad5021cc76f.png" width="250">

Reply:
hmm interesting, I didn't know about `make_dataclass`. However, doing something like this, we'd want it to be `frozen=True` so that users can't change the values, which is easy enough to implement. But for the docs build you get something like this

![image](https://user-images.githubusercontent.com/61475504/204545926-2dbe080d-74e7-4335-9b22-ad2ab1339dc7.png)